workshop/content/20-typescript/30-hello-cdk/300-cdk-watch.md

```diff
@@ -0,0 +1,298 @@
+ ✅  CdkWorkshopStack
+
+✨  Deployment time: 66.82s
+
+Stack ARN:
+arn:aws:cloudformation:REGION:ACCOUNT-ID:stack/CdkWorkshopStack/STACK-ID
+
+✨  Total time: 72.82s
+```
+
+The exact time will vary but we should get a pretty good idea of how long a normal
+deployment takes!
+
+## Hotswap deployments
+
+{{% notice info %}} This command deliberately introduces drift in CloudFormation 
+stacks in order to speed up deployments. For this reason, only use it for 
+development purposes. Never use hotswap for your production deployments!
+{{% /notice %}}
+
+We can speed up that deployment time with `cdk deploy --hotswap`, which will
+assess whether a hotswap deployment can be performed instead of a CloudFormation
+deployment. If possible, the CDK CLI will use AWS service APIs to directly make
+the changes; otherwise it will fall back to performing a full CloudFormation
+deployment.
+
+Here, we will use `cdk deploy --hotswap` to deploy a hotswappable change to your 
+AWS Lambda asset code.
+
+## Timing `cdk deploy --hotswap`
+
+Let's change the lambda code in `lambda/hello.js` another time:
+
+{{<highlight js "hl_lines=6">}}
+exports.handler = async function(event) {
+  console.log("request:", JSON.stringify(event, undefined, 2));
+  return {
+    statusCode: 200,
+    headers: { "Content-Type": "text/plain" },
+    body: `Good Afternoon, CDK! You've hit ${event.path}\n`
+  };
+};
+{{</highlight>}}
+
+Now, let's run `cdk deploy --hotswap`:
+
+```
+cdk deploy --hotswap
+```
+
+The output will look something like this:
+
+```
+✨  Synthesis time: 6.44s
+
+⚠️ The --hotswap flag deliberately introduces CloudFormation drift to speed up deployments
+⚠️ It should only be used for development - never use it for your production Stacks!
+
+CdkWorkshopStack: deploying...
+✨ hotswapping resources:
+   ✨ Lambda Function 'CdkWorkshopStack-HelloHandler2E4FBA4D-tEZTcXqG8YYe'
+✨ Lambda Function 'CdkWorkshopStack-HelloHandler2E4FBA4D-tEZTcXqG8YYe' hotswapped!
+
+ ✅  CdkWorkshopStack
+
+✨  Deployment time: 3.07s
+
+Stack ARN:
+arn:aws:cloudformation:REGION:ACCOUNT-ID:stack/CdkWorkshopStack/STACK-ID
+
+✨  Total time: 9.51s
+```
+
+Wow, deploying a hotswapped change took 3 seconds, while a full deployment took 67 seconds! 
+But take a look and read the warning message thoroughly - it's important!:
+
+```
+⚠️ The --hotswap flag deliberately introduces CloudFormation drift to speed up deployments
+⚠️ It should only be used for development - never use it for your production Stacks!
+```
+
+## Did the code actually change?
+
+Wow that was fast. Did the code actually change? Let's go to the AWS Lambda Console and
+double check!
+
+1. Open the [AWS Lambda
+   Console](https://console.aws.amazon.com/lambda/home#/functions) (make sure
+   you are in the correct region).
+
+    You should see our function:
+
+    ![](./lambda-1.png)
+
+2. Click on the function name to go to the console.
+
+3. The code should be loaded onto the screen. Did your change show up?
+
+    ![](./lambda-5.png)
```

Comment:
You might consider also testing that the function changed by calling it, and observing the result it returned. That's super minor though, this is fine too if you want to keep it as-is.

Reply:
We do exactly this (test the result) in the previous page where we set up the lambda. So I think it is fine to keep it as-is here; feels like observing the code change is evidence enough.